### PR TITLE
Update contour to 1.27.0 and Kubernetes to 1.28

### PR DESCRIPTION
**Pull request by yokaze**
Signed-off-by: Daichi Sakaue <daichi-sakaue@cybozu.co.jp>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -89,3 +158,40 @@ lint: ## Run lint tools
...
+	$(eval latest_tag := $(shell wget -O - https://quay.io/api/v1/repository/cybozu/$1/tag/ | jq -r '.tags[] | .name' | awk '/.*\..*\./ {print $$1; exit}'))
+endef
+
+# usage: upstream-tag 1.2.3.4
+# do not indent because it appears on output
+define upstream-tag
+$(shell echo $1 | sed -E 's/^(.*)\.[[:digit:]]+$$/v\1/')
+endef
+
+# usage update-version OWNER/REPO VAR MAJOR
```
Comment (terassyi) [inline Makefile:178]:
```suggestion
# usage update-version OWNER/REPO VER MAJOR
```

`VAR` seems to mean "version", so I think this should be `VER`.

File: Makefile [hunk 2]
```diff
@@ -69,6 +79,65 @@ build: ## Build manager binary
...
+	$(call update-trusted-action,actions/checkout,$(ACTIONS_CHECKOUT_VERSION))
+	$(call update-trusted-action,actions/create-release,$(ACTIONS_CREATE_RELEASE_VERSION))
+	$(call update-trusted-action,actions/setup-go,$(ACTIONS_SETUP_GO_VERSION))
+
+.PHONY: maintenance
+maintenance: ## Update dependent manifests
+	$(MAKE) update-actions
+	$(MAKE) download-crds
+
+.PHONY: list-actions
```
Comment (terassyi) [inline Makefile:135]: This target isn't used in other targets and doesn't appear in `maintenance.md`.
When are you supposed to use this?

Comment (yokaze) [reply Makefile:135]: @terassyi 
This is a small usable function to check the completeness of the update script.
It is not used unless we're going to use a new Action.

If you suspect it is not needed, I'll move it to my `dotfiles`. How do you think?

Comment (terassyi) [reply Makefile:135]: It is good for me to keep this in the file.
I just thought it might be a copy paste mistake. 🙇‍♂️

File: Makefile [hunk 3]
```diff
@@ -89,3 +158,40 @@ lint: ## Run lint tools
...
+endef
+
+# usage update-version OWNER/REPO VERSION MAJOR
+define update-version
+	$(call get-latest-gh,$1)
+	NEW_VERSION=$$(echo $(latest_gh) | if [ -z "$3" ]; then cut -b 2-; else cut -b 2; fi); \
+	sed -i -e "s/$2 := .*/$2 := $${NEW_VERSION}/g" Makefile.versions
+endef
+
+# usage update-version-quay NAME VAR
```
Comment (terassyi) [inline Makefile:185]:
```suggestion
# usage update-version-quay NAME VERSION
```

**Review by terassyi — CHANGES_REQUESTED** [review]

File: Makefile (hunk 3 above)
Comment (yokaze) [reply Makefile:185]: For `update-version` and `update-version-quay`, the parameter is a `VAR` (variable name).
It receives a repository name and `VAR` name, and write a record in `Makefile.versions`.

```make
# usage update-version OWNER/REPO VAR MAJOR
define update-version
	$(call get-latest-gh,$1)
	NEW_VERSION=$$(echo $(latest_gh) | if [ -z "$3" ]; then cut -b 2-; else cut -b 2; fi); \
	sed -i -e "s/$2 := .*/$2 := $${NEW_VERSION}/g" Makefile.versions
endef
```

On the other hand, `update-trusted-action` receives a specific version and overwrites GitHub Actions manifests.

```make
# usage update-trusted-action OWNER/REPO VERSION
define update-trusted-action
	for i in $(shell ls $(WORKFLOWS_DIR)); do \
		$(YQ) -i '(.. | select(has("uses")) | select(.uses | contains("$1"))).uses = "$1@v$2"' $(WORKFLOWS_DIR)/$$i; \
	done
endef
```

Therefore, their calling conventions are different:
```make
$(call update-version,actions/checkout,ACTIONS_CHECKOUT_VERSION,1)
$(call update-version-quay,cert-manager,CERT_MANAGER_VERSION)

# The version is substituted for update-trusted-action
$(call update-trusted-action,actions/checkout,$(ACTIONS_CHECKOUT_VERSION))
```

**Review by terassyi — APPROVED** [review]
LGTM

**Pull request merged**